File: 1411-Number of Ways to Paint N × 3 Grid.py
class Solution(object):
    def numOfWays(self, n):
        """
        :type n: int
        :rtype: int
        """
# https://www.youtube.com/watch?v=uId0P2zAHKk&ab_channel=LeadCoding
        same = diff = 6
        modulo = 10**9 + 7
        
        for i in range(1,n):
            tmp = same
            same = (2*diff + 3*tmp)%modulo
            diff = (2*tmp + 2*diff)%modulo
        
        return (same+diff)%modulo

File: JavaScript/1411-Number of Ways to Paint N × 3 Grid.js
/**
 * @param {number} n
 * @return {number}
 */
var numOfWays = function(n) {
    let same = 6, diff = 6;
    let modulo = Math.pow(10,9)+ 7;
    
    for (let i=1; i<n; i++){
        let tmp = same;
        same = (2*diff + 3*same)%modulo;
        diff = (2*diff + 2*tmp)%modulo;
    }
    
    return (same + diff)%modulo;
};